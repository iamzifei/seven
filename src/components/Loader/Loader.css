.overlay-loader {
  position: fixed; /* Sit on top of the page content */
  width: 100%; /* Full width (cover the whole page) */
  height: 100%; /* Full height (cover the whole page) */
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(35, 50, 56, 0.54); /* Black background with opacity */
  z-index: 999999999; /* Specify a stack order in case you're using a different order for other elements */
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.loader {
  width: 72px;
  height: 72px;
  /* margin: 0 auto; */
  margin-top: 50px;
  position: relative;
  -webkit-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
}
.loader .cube {
  position: relative;
  -webkit-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
}
.loader .cube {
  float: left;
  width: 50%;
  height: 50%;
  position: relative;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}
.loader .cube:before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #007fad;
  -webkit-animation: foldthecube 2.4s infinite linear both;
  animation: foldthecube 2.4s infinite linear both;
  -webkit-transform-origin: 100% 100%;
  -ms-transform-origin: 100% 100%;
  transform-origin: 100% 100%;
}
.loader .c2 {
  -webkit-transform: scale(1.1) rotateZ(90deg);
  transform: scale(1.1) rotateZ(90deg);
}
.loader .c3 {
  -webkit-transform: scale(1.1) rotateZ(180deg);
  transform: scale(1.1) rotateZ(180deg);
}
.loader .c4 {
  -webkit-transform: scale(1.1) rotateZ(270deg);
  transform: scale(1.1) rotateZ(270deg);
}
.loader .c2:before {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}
.loader .c3:before {
  -webkit-animation-delay: 0.6s;
  animation-delay: 0.6s;
}
.loader .c4:before {
  -webkit-animation-delay: 0.9s;
  animation-delay: 0.9s;
}
@keyframes foldthecube {
  0%,
  10% {
    -webkit-transform: perspective(140px) rotateX(-180deg);
    transform: perspective(140px) rotateX(-180deg);
    opacity: 0;
  }
  25%,
  75% {
    -webkit-transform: perspective(140px) rotateX(0deg);
    transform: perspective(140px) rotateX(0deg);
    opacity: 1;
  }
  90%,
  100% {
    -webkit-transform: perspective(140px) rotateY(180deg);
    transform: perspective(140px) rotateY(180deg);
    opacity: 0;
  }
}
