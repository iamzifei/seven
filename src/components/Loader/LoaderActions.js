export const PAGE_LOAD_START = "PAGE_LOAD_START";
export const PAGE_LOAD_END = "PAGE_LOAD_END";

export const startLoading = () => {
  return {
    type: PAGE_LOAD_START
  };
};

export const endLoading = () => {
  return {
    type: PAGE_LOAD_END
  };
};
