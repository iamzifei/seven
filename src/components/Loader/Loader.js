import React from "react";

import "./Loader.css";

const Loader = () => (
  <div className="loader">
    <div className="cube c1" />
    <div className="cube c2" />
    <div className="cube c4" />
    <div className="cube c3" />
  </div>
);

export default Loader;
