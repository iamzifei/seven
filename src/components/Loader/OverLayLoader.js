import React from "react";
import Loader from "./Loader";

import "./Loader.css";

const OverLayLoader = () => (
  <div className="overlay-loader">
    <Loader />
  </div>
);

export default OverLayLoader;
