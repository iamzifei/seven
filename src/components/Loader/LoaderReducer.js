// Import Actions
import * as actions from "./LoaderActions";

// Initial State
const initialState = {
  pageLoading: false
};

// Reducer
const Reducer = (state = initialState, action) => {
  const { type } = action;
  switch (type) {
    case actions.PAGE_LOAD_START:
      return {
        ...state,
        pageLoading: true
      };
    case actions.PAGE_LOAD_END:
      return {
        ...state,
        pageLoading: false
      };
    default:
      return state;
  }
};

export default Reducer;
