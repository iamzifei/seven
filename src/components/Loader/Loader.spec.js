import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Loader from "./Loader";

describe("Loader testing", () => {
  const enzymeWrapper = shallow(<Loader />);
  it("renders with right snapshot", () => {
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  });

  it("renders with right classname", () => {
    expect(enzymeWrapper.hasClass("loader")).toBe(true);
    expect(enzymeWrapper.children("div").length).toBe(4);
    expect(enzymeWrapper.children("div.cube").length).toBe(4);
    expect(enzymeWrapper.children("div.c1").length).toBe(1);
    expect(enzymeWrapper.children("div.c2").length).toBe(1);
    expect(enzymeWrapper.children("div.c3").length).toBe(1);
    expect(enzymeWrapper.children("div.c4").length).toBe(1);
  });
});
