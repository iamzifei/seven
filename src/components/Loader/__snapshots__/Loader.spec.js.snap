// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader testing renders with right snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="cube c1"
  />
  <div
    className="cube c2"
  />
  <div
    className="cube c4"
  />
  <div
    className="cube c3"
  />
</div>
`;
